### PR TITLE
Sail overlay: map-truth boat traversability and validation

### LINKED ISSUE #388: Sail overlay: map-truth boat traversability and validation
## Problem Statement

World Builder **meander refine** and corridor attraction were built to improve connected, boat-viable river networks on the final **landmass** map. In practice, hydrology **validation checks** and the generation report still read **simulation hydrology** (`riverGraph` edges from the pre-refine settled centerline). The map can show physically connected rivers—especially after **meander refine** bridges nearby corridors—while validation reports disconnected “navigable” networks and **rejection sampling** may reject worlds that look sail-viable on screen.

The product intent is that boat **traversable water** follows what the player sees: if you can draw a path through the water on the final map without leaving the sailing surface, you can sail between those points. Shoreline outline pixels and thin gaps between waterways break a naive “blue pixel” test; the correct operational layer is a derived **Sail overlay** (bright pink), not raw biome tint or simulation graph edges.

See [ADR 0010](docs/adr/0010-world-builder-sail-overlay-traversability.md) and glossary terms in `world-builder/CONTEXT.md` (**Sail overlay**, **Sailable water**, **Coastal river access**, **Coast-to-interior sailing path**).

## Solution

Introduce a **Sail overlay** computed on demand from the final display water union (ocean + lakes + painted river corridors), processed through fixed blur/high-pass constants to merge waterways, include a usable waterfront sliver for launch/landings, and expose it as a bright-pink map overlay toggle (**Sail**).

Rewire hydrology **validation checks** and the generation report to use **Sail overlay** connectivity (8-connected paths through set pixels) instead of simulation graph edge counts. Rename user-facing validation and report labels to sail-native language while keeping internal `enforce*` option keys stable for saved settings.

**Meander refine** remains optional; checks describe the generated map as shown, not a counterfactual with refine enabled—but when refine runs, its bridges must affect **Sail overlay** because they affect painted corridors.

## User Stories

1. As a world author, I want boat traversability to match the final map I see, so that connected rivers on screen count as sail-connected in validation.
2. As a world author, I want a **Sail** overlay toggle on the map, so that I can see exactly where simulated people can launch, land, and sail.
3. As a world author, I want the **Sail overlay** rendered bright pink, so that it is visually distinct from terrain and resource overlays.
4. As a world author, I want the **Sail overlay** to include rivers, lakes, and ocean as one sailing surface, so that I can trace boat routes across all water types.
5. As a world author, I want nearby waterways to connect in the **Sail overlay** after blur processing, so that thin gaps and outline pixels do not falsely block sailing between connected bodies.
6. As a world author, I want a narrow waterfront sliver around waterways in the **Sail overlay**, so that shore access for launching and landing ships is represented without requiring exact blue-pixel alignment.
7. As a world author, I want two points to count as sail-connected when an 8-connected path through the **Sail overlay** never crosses unset pixels, so that diagonal shore contacts work correctly.
8. As a world author, I want **meander refine** bridges to improve **Sail overlay** connectivity when enabled, so that the refine pass achieves its purpose of better boat networks.
9. As a world author, I want validation to still pass honestly when **meander refine** is off, so that I am not forced to enable an expensive pass to get a truthful report.
10. As a world author, I want the validation sidebar row **Sailable water** to reflect meaningful connected sailing area, so that fragmented puddles do not pass as viable networks.
11. As a world author, I want **Sailable water** to require the largest connected **Sail overlay** component to meet a threshold, so that tiny isolated water blobs fail appropriately.
12. As a world author, I want a separate **Coastal river access** validation row, so that I can see whether any river or lake meets the ocean on the overlay.
13. As a world author, I want a **Coast-to-interior sailing path** validation row, so that I can see whether inland water is reachable from the sea by boat.
14. As a world author, I want generation report hydrology stats to show sail metrics instead of graph edge counts, so that the numbers match the **Sail overlay** contract.
15. As a world author, I want legacy “navigable edges” and graph mouth counts removed from the default generation report, so that outdated metrics do not mislead me.
16. As a world author, I want internal enforce toggles (`enforceNavigableRiverQuota`, etc.) to keep working with saved settings, so that my persisted generation options are not broken by label renames.
17. As a world author, I want user-facing validation labels to say **Sailable water**, **Coastal river access**, and **Coast-to-interior sailing path**, so that the sidebar language matches the sailing model.
18. As a world author, I want the **Sail overlay** derived on demand rather than stored on the **world document**, so that validation and display always agree on the same algorithm.
19. As a world author, I want blur and high-pass parameters fixed in v1, so that validation outcomes do not drift from what the overlay shows.
20. As a developer maintaining World Builder, I want a single pure derive function for the **Sail overlay** mask used by both renderer and validation, so that there is one source of truth.
21. As a developer, I want **simulation hydrology** to remain available for drainage physics and tuning, so that presentation sailing checks do not break flow solves or settled centerline export.
22. As a developer, I want tests that previously enforced simulation-first validation invariance (#358, #365) revised, so that CI reflects the ADR 0010 contract.
23. As a developer, I want seam-level integration tests proving meander-refine bridges move **Sail overlay** metrics, so that the original connectivity bug cannot regress.
24. As a GM reading validation failures, I want **rejection sampling** to reject worlds with inadequate **Sail overlay** connectivity when enforce toggles are on, so that generated **landmasses** support future **maritime reach** and **trade route** work.
25. As a future simulation author, I want upstream/downstream sailing cost deferred, so that v1 can ship map-truth connectivity without full river navigation physics.

### Sail overlay derivation (new core module)

- Add a pure derive pipeline: build binary water union from ocean (below sea level), `lakeMask`, and painted `riverCorridorMask` (post-refine, post-paint); apply fixed-radius blur; apply high-pass threshold; emit set/unset mask suitable for connectivity analysis and renderer tinting.
- Constants (blur radius, high-pass threshold) are fixed pipeline values in v1—not user sliders, not persisted on the **world document**.
- Same derive entry point called from validation/metrics and from the map renderer overlay path.

### Sail connectivity metrics (new core module)

- **8-connected** connected-component labeling on the derived mask.
- **Sailable water metric:** largest component cell count (compare to configured minimum threshold; maps to `navigableRiverQuota` enforce option internally).
- **Coastal river access:** boolean or count—overlay component or path connects ocean cells to inland river/lake water (maps to `coastMouth` check id internally).
- **Coast-to-interior sailing path:** longest or qualifying inland reach from coast through set pixels (maps to `coastConnectedNavigablePath` internally).

### Validation layer changes

- Rewire `runGeographyValidationChecks` and `computeHydrologyMetrics` (or successor) to consume **Sail overlay** metrics for the three sailing checks instead of `riverGraph` navigable edges.
- Update validation row display labels via landmass validation contracts / page model formatters—check ids and `enforce*` keys remain stable (`navigableRiverQuota`, `coastMouth`, `coastConnectedNavigablePath`).
- Remove or stop populating user-facing generation report fields for navigable edge count, graph mouth count, and navigable km from centerline graph; replace with sail metric fields.

### Renderer and UI

- Add **Sail** to map overlay toggles (same visibility pattern as resource overlays—status bar or overlay panel per existing World Builder UX).
- Render derived mask as bright pink raster overlay on the map viewport.
- Update **meander refine** generation control tooltip to describe connecting waterways for sailing (map-truth), not vague display smoothing.

### Simulation hydrology (unchanged role for physics)

- `simulationRiverMask` byte invariance under presentation toggles may remain for simulation export (#358); it no longer gates sailing validation.
- `buildRiverGraph` on settled mask may remain for diagnostics or future logistics—not for default sailing validation.

### Documentation

- ADR 0010 already recorded; update seam docs (`SIMULATION-VS-PRESENTATION-HYDROLOGY.md`, `ARCHITECTURE-SEAMS.md`, validation stage spec) to reflect sailing checks on **Sail overlay**.
- Glossary in `world-builder/CONTEXT.md` already updated in grill session.

## Testing Decisions

**What makes a good test:** assert externally visible contracts—validation row status, generation report metric values, overlay connectivity outcomes, pipeline rejection behavior—not blur kernel internals or copy strings in unit tests unless testing stable API labels via contracts.

**Preferred seams (highest first):**

1. **Full derived geography pipeline + validation** — generate with/without **meander refine**; assert **Sail overlay**-backed checks change when presentation bridges connect waterways; assert simulation mask invariance tests are removed or replaced per ADR 0010.
2. **`runGeographyValidationChecks` integration** — synthetic world document fixtures with divergent presentation vs simulation masks; sailing checks follow overlay, not graph edges.
3. **Sail derive + metrics pure modules** — small grid fixtures: water union in → component sizes and coastal path lengths out (behavioral, no snapshot of marketing copy).
4. **Renderer overlay seam** — viewport applies Sail visibility toggle and pink raster (follow `resourceRasterOverlayRefresh` / overlay visibility test patterns).
5. **Page model / generation report display** — hydrology stats expose sail fields; graph-edge counts absent from default display model.

**Prior art:** `runGeographyValidationChecks.test.js`, `computeHydrologyMetrics.test.js`, `hydrologySubsteps.test.js` (presentation vs simulation mask tests—revise), `resourceOverlayState.test.js`, `worldBuilder.integration.test.js`.

## Out of Scope

- Upstream vs downstream sailing cost (future: elevation incline + flow direction).
- User-adjustable blur/high-pass sliders (display-only or generation).
- Persisting `sailTraversibilityMask` on the **world document**.
- Renaming internal `enforce*` option keys or validation `checkId` strings.
- Land **movement cost** graph (non-sail **logistics pass**)—this PRD is sailing/water only.
- Forcing **meander refine** on by default or auto-enabling when sail checks fail.
- Removing simulation hydrology or Hack's-law-style diagnostics from codebase entirely (only from default user-facing report).

## Further Notes

- Root bug discovered when **meander refine** was disabled in UI but corridor attraction still ran under the old skip logic (since fixed UI-side); the deeper issue is validation reading simulation graph while map shows presentation connectivity.
- **Testing seams:** confirm during implementation that highest-seam pipeline tests are sufficient before adding lower-level blur unit tests only where needed for edge cases.
- Related ADR: [0010-world-builder-sail-overlay-traversability.md](docs/adr/0010-world-builder-sail-overlay-traversability.md)

### PR DESCRIPTION
**Pull request by enmaku**
## Summary

- Adds a **Sail overlay** derived on demand from final display water (ocean + lakes + painted river corridors) via blur/high-pass, exposed as a bright-pink map toggle and used as the single source of truth for boat traversability.
- Rewires hydrology validation and the generation report to use **Sail overlay** connectivity metrics (8-connected paths) instead of simulation `riverGraph` edge counts, with sail-native user-facing labels while keeping internal `enforce*` keys stable.
- Records ADR 0010 and updates World Builder glossary, seam docs, and tests to match the presentation-first sailing contract.

Closes #388

## Test plan

- [ ] Generate a world with **meander refine** on and off; confirm connected waterways on the map pass **Sailable water** / **Coastal river access** / **Coast-to-interior sailing path** when the Sail overlay shows a path.
- [ ] Toggle the **Sail** overlay on the map viewport; confirm bright-pink raster aligns with navigable water regions.
- [ ] Confirm generation report hydrology stats show sail metrics (not legacy navigable edge / graph mouth counts).
- [ ] `npm test` — sail derive/metrics, validation checks, renderer overlay, and pipeline seam tests pass.

### CODE REVIEW

File: world-builder/core/buildGenerationReport.js [hunk 1]
```diff
@@ -95,7 +102,9 @@ export function buildGenerationReport({
     arableRaster,
     saltNodes,
     metalNodes,
-  })
+  }
+  const { metrics: sailMetrics } = resolveSailOverlayFromSlice(validationSlice)
+  const validationRows = runGeographyValidationChecks(validationSlice)
```
Comment (enmaku) [inline world-builder/core/buildGenerationReport.js:107]: [structure] `resolveSailOverlayFromSlice` runs here, then `runGeographyValidationChecks(validationSlice)` derives the same mask+metrics again internally. That's two full blur passes per generation report on every validation run. Pass `{ mask, metrics }` on the slice (or have validation accept an optional precomputed snapshot) and delete the duplicate derive.

File: world-builder/core/validation/runGeographyValidationChecks.js [hunk 2]
```diff
@@ -95,30 +99,30 @@ export function runGeographyValidationChecks(slice) {
     throw new Error('riverNetwork or hydrologyMetrics required for geography validation')
   }
 
-  const navigableEdges = selectNavigableRiverEdges(graph.edges)
-  const minNavigable = minNavigableRiverEdgesForGrid(gridWidth)
-  const navigablePass = navigableEdges.length >= minNavigable
+  const { mask: sailMask, metrics: sailMetrics } = resolveSailOverlayFromSlice(slice)
```
Comment (enmaku) [inline world-builder/core/validation/runGeographyValidationChecks.js:102]: [structure] This is the second derive when called from `buildGenerationReport`. Consider `slice.sailOverlay ?? resolveSailOverlayFromSlice(slice)` so callers that already computed the snapshot don't pay twice.

File: world-builder/renderer/worldBuilderMapViewportModel.js [hunk 3]
```diff
@@ -76,3 +79,27 @@ export function resolveMetalsOverlayDrawn(visibility, worldDocument) {
...
+ */
+export function resolveSailRasterLayerVisible(visibility, worldDocument) {
+  if (!isResourceOverlayVisible(visibility, 'sail')) {
+    return false
+  }
+  const { gridWidth, gridHeight, fields, lakeMask, riverCorridorMask } = worldDocument
+  if (!fields?.elevation) {
+    return false
+  }
+  const mask = deriveSailOverlayMask({
```
Comment (enmaku) [inline world-builder/renderer/worldBuilderMapViewportModel.js:96]: [architecture] Visibility resolution runs the full blur/high-pass pipeline just to check `.some()`, then `buildSailOverlayRgba` runs it again on refresh. A cheap precheck (raw water union has any cell) or reusing the mask from the refresh seam would keep derive locality in one place.

File: world-builder/core/sail/resolveSailMetricsFromSlice.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  return resolveSailOverlayFromSlice(slice).metrics
+}
+
+/**
+ * @param {Uint8Array} mask
+ * @param {number} gridWidth
+ * @param {number} gridHeight
+ * @returns {import('../types.js').MapFocus}
+ */
+export function findSailOverlayFocus(mask, gridWidth, gridHeight) {
```
Comment (enmaku) [inline world-builder/core/sail/resolveSailMetricsFromSlice.js:68]: [architecture] `findSailOverlayFocus` reimplements 8-connected component labeling that `collectConnectedComponents` already provides in `computeSailMetrics`. Reuse the canonical helper and pick the centroid of the largest component — keeps graph-walk knowledge in one module.

File: world-builder/core/buildGenerationReport.js [hunk 5]
```diff
@@ -166,6 +176,9 @@ export function buildGenerationReport({
       slopeAreaConcavitySamples: metrics.slopeAreaConcavitySamples,
       parallelStrandRatio: metrics.parallelStrandRatio,
       coastConnectedNavigablePathLength: metrics.coastConnectedNavigablePathLength,
```
Comment (enmaku) [inline world-builder/core/buildGenerationReport.js:178]: [architecture] User-facing report/UI dropped graph `navigableEdgeCount` / mouth metrics, but `hydrology` still exports `coastConnectedNavigablePathLength` from the simulation graph beside sail-native `coastToInteriorSailPathLength`. Consider dropping or renaming the graph field so the hydrology object doesn't straddle two traversability seams.

**Review by enmaku — COMMENTED** [review]
## Verdict
COMMENT — Would approve, but posting COMMENT (own PR). ADR 0010 is implemented coherently: one derive function feeds validation, report, and renderer; seam docs and tests match the new map-truth contract. A few locality and duplication issues are worth fixing before merge, none blocking correctness.

## Thermo-nuclear (diff)
- **Duplicate sail derive on the validation hot path:** `buildGenerationReport` calls `resolveSailOverlayFromSlice`, then `runGeographyValidationChecks` derives the same mask again. On a 256² grid that's two full blur passes per report — the obvious code-judo move is to pass the snapshot through the slice once.
- **Renderer visibility preflight runs the full pipeline:** `resolveSailRasterLayerVisible` derives the entire mask only to call `.some()`, then `buildSailOverlayRgba` derives it again on refresh.
- **Removed `elevationPriorPassRates.test.js` without a sail-era replacement:** `countRiverValidationPasses` still exists and now reads sail-backed rows, but nothing guards the 50-seed batch pass-rate floor anymore. Worth restoring with updated thresholds.

## Architecture (PR-scoped)
- **`findSailOverlayFocus` is a second 8-connected CC implementation** alongside `collectConnectedComponents` in `computeSailMetrics` — shallow duplication across the sail module; reuse would concentrate graph-walk locality.
- **`report.hydrology` mixes graph and sail path lengths** (`coastConnectedNavigablePathLength` vs `coastToInteriorSailPathLength`) in one object after UI dropped graph metrics — seam leakage for future readers.
- **Sail raster refresh seam does derive → visibility → derive** — three entry points (`resolveSailRasterLayerVisible`, `buildSailOverlayRgba`, validation) all call `deriveSailOverlayMask` independently; a cached snapshot on the overlay refresh context would deepen the renderer adapter.

## Unanchored notes
- Deleting the #365 presentation-invariance integration test is correct given ADR 0010; the replacement corridor-bridge test is the right seam assertion.
- `REFERENCE_MIN_SAILABLE_WATER_CELLS = 48` on reference grid — no seed-batch regression test yet to catch tuning drift (ties to deleted pass-rate tests).
